packages/itwin/tree-widget/src/e2e-tests/CategoriesTree.test.ts

```diff
@@ -112,5 +97,102 @@ test.describe("Categories tree", () => {
       await page.locator(".tree-header-button-dropdown-container").waitFor();
       await takeScreenshot(page, treeWidget);
     });
+
+    test("shows outlines when focused using keyboard", async ({ page }) => {
+      // click to focus on node
+      const node = locateNode(treeWidget, "Equipment", 1);
+      await node.click();
+
+      // focus on checkbox using keyboard
+      await page.keyboard.press("Tab");
+
+      // ensure checkbox is focused
+      const checkbox = node.locator(".tw-tree-node-checkbox");
+      await expect(checkbox).toBeFocused();
+
+      await takeScreenshot(page, node, { expandBy: { top: 10, bottom: 10 } });
+
+      // shrink panel
+      await expandStagePanel(page, "right", -100);
+
+      // re-focus on checkbox after resizing the panel
+      await node.click();
+      await page.keyboard.press("Tab");
+
+      // scroll to the right side
+      await scrollTree(page, 10000, 0);
+      await takeScreenshot(page, node, { boundingComponent: treeWidget, expandBy: { top: 10, bottom: 10 } });
+
+      // expand panel
+      await expandStagePanel(page, "right", 100);
+
+      // scroll to origin to avoid flakiness due to auto-scroll
+      await scrollTree(page, -10000, -10000);
+
+      // re-focus on checkbox after resizing the panel
+      await node.click();
+      await page.keyboard.press("Tab");
+
+      // focus on expander
+      await page.keyboard.press("Tab");
+      const expander = node.getByLabel("Expand");
+      await expect(expander).toBeFocused();
+
+      await takeScreenshot(page, node, { expandBy: { top: 10, bottom: 10 } });
+
+      // navigate back to focus on the already selected node
+      await page.keyboard.press("ArrowUp");
+      await page.keyboard.press("ArrowDown");
+
+      await expect(node).toBeFocused();
+
+      await takeScreenshot(page, node, { expandBy: { top: 10, bottom: 10 } });
+
+      // click on a different node to avoid showing filtering buttons due to hover
+      const hexNode = locateNode(treeWidget, "E-HEX");
+      await hexNode.click();
+
+      // focus back on the node we want to filter
+      await page.keyboard.press("ArrowDown");
+
+      // Focus on apply filter button
+      await page.keyboard.press("Tab");
+      await page.keyboard.press("Tab");
+      await page.keyboard.press("Tab");
+
+      await takeScreenshot(page, node, { expandBy: { top: 10, bottom: 10 } });
+
+      // open filtering dialog
+      await page.keyboard.press("Enter");
+
+      await locateInstanceFilter(page).waitFor();
+      await selectPropertyInDialog(page, "Code");
+      await selectOperatorInDialog(page, "Equal");
+      await selectValueInDialog(page, "Equipment - Insulation");
+
+      await page.getByRole("button", { name: "Apply" }).click();
+
+      // bring focus on the node
+      await hexNode.click();
+      await page.keyboard.press("ArrowDown");
```

Comment:
We're testing if its possible to apply a filter with keyboard. Where does the focus go after applying filter? Can the test be updated to use only keyboard and no `click`?

Reply:
The focus seems to disappear but it can be shift + tabbed back. Removed the `.click`, though did not change the rest of the test.

Reply:
Do you think the requirement to shift+tab to focus filtered node after applying a filter on it is intuitive? Maybe we should do that automatically?

Reply:
So it seems after applying filter the focus is kept on the apply filter button, however it is not being shown. Shift+tab causes it to go back to the clear filter button (and show the outline). It is not very intuitive, and seems like the focus is completely lost when a filter is cleared, so I will look into these issues. I assume the focus should be kept on the apply filter button after closing the dialog, but do you think the focus should be transferred to the node after clearing the filter or the apply filter button?

Reply:
The outline issue turned out to be caused by the fact that the filtering dialog is used via a mouse.

I opened a PR for switching focus to "Apply filter" button when the filter is cleared - https://github.com/iTwin/presentation/pull/649. The changes in that PR should not cause any changes in this one so this comment can be resolved (unless we want to additionally include a test for ensuring focus is switched when a filter is cleared).

Reply:
I think improving the `shows outlines when focused using keyboard` tests to check if the focus moves as expected would be a good idea. E.g. for the Models tree, I'd make the following change:

```ts
      await page.getByRole("button", { name: "Apply" }).click();

      // EXPECT "Apply filter" BUTTON TO BE FOCUSED
      // TAKE SCREENSHOT

      // navigate to clear filter button
      await page.keyboard.press("Shift+Tab");

      // scroll to origin to avoid flakiness due to auto-scroll
      await scrollTree(page, -10000, -10000);
      await takeScreenshot(page, node, { expandBy: { top: 10, bottom: 10 } });

      // PRESS "Enter" TO CLICK THE "Clear filter" BUTTON
      // EXPECT "Apply filter" BUTTON TO BE FOCUSED

      // DON'T NEED THESE ANYMORE:
      // // navigate to apply filter button
      // await page.keyboard.press("Tab");
      // 
      // // scroll to origin to avoid flakiness due to auto-scroll
      // await scrollTree(page, -10000, -10000);
      // await takeScreenshot(page, node, { expandBy: { top: 10, bottom: 10 } });
```

Also, we're repeating this piece a lot:
```ts
      // scroll to origin to avoid flakiness due to auto-scroll
      await scrollTree(page, -10000, -10000);
      await takeScreenshot(page, node, { expandBy: { top: 10, bottom: 10 } });
```
Maybe it could be extracted into a function?

Reply:
Updated the tests to verify the focus moves correctly. Rather than creating a separate function I added an option to `takeScreenshot` to reset the scroll.

Reply:
> Updated the tests to verify the focus moves correctly. 

The tests are now taking 1 screenshot less than before - I believe that last tests' screenshot needs to be removed from screenshots dir.

> Rather than creating a separate function I added an option to takeScreenshot to reset the scroll.

I like that, thanks.

Reply:
Right, removed.